lib/HLSL/DxilValidation.cpp

```diff
@@ -3988,6 +3988,114 @@ static void ValidateMetadata(ValidationContext &ValCtx) {
   }
 
   const hlsl::ShaderModel *SM = ValCtx.DxilMod.GetShaderModel();
+  // validate that any wavesize tags don't appear outside their expected shader
+  // models. Validate only 1 tag exists per entry point.
+  // Don't do this validation if the shader is non-compute
```

Comment:
It's an extremely small nit, but a space between the comments describing the next block of code and the comment describing the single conditional line might make it a bit less confusing. 

As long and as nested as this is though, perhaps a separate function call would be useful to keep all the wave size range code in the same place